### PR TITLE
IGNITE-12821 Add check size into validate_indexes

### PR DESCRIPTION
**Pull request by tkalkirill**

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerAbstractTest.java [hunk 1]
```diff
@@ -185,18 +188,17 @@ protected boolean idleVerifyRes(Path p) {
         if (sslEnabled())
             cfg.setSslContextFactory(GridTestUtils.sslFactory());
 
-        DataStorageConfiguration memCfg = new DataStorageConfiguration()
+        DataStorageConfiguration dsCfg = new DataStorageConfiguration()
+            .setWalMode(WALMode.LOG_ONLY)
```
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerAbstractTest.java:192]: I'll leave it.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java [hunk 2]
```diff
@@ -2906,13 +2908,11 @@ public void remove(GridCacheContext cctx, CacheDataRow row)
      * @return Descriptors.
      */
     public Collection<GridQueryTypeDescriptor> types(@Nullable String cacheName) {
-        Collection<GridQueryTypeDescriptor> cacheTypes = new ArrayList<>();
+        Collection<GridQueryTypeDescriptor> cacheTypes = newSetFromMap(new IdentityHashMap<>());
```
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java:2911]: Just can be duplicates, look how this Map is filled, start from that line
https://github.com/apache/ignite/blob/5a8b0cc4894a99ef4b22b988b9aa30859485b1b9/modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java#L1833
`
types.put(typeId, desc);

if (altTypeId != null)
    types.put(altTypeId, desc);
`

File: modules/core/src/main/java/org/apache/ignite/internal/visor/verify/VisorValidateIndexesTaskArg.java [hunk 3]
```diff
@@ -37,17 +39,20 @@
     private Set<String> caches;
 
     /** Check first K elements. */
-    private int checkFirst;
-
-    /** Check CRC */
-    private boolean checkCrc;
+    private int checkFirst = -1;
```
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/verify/VisorValidateIndexesTaskArg.java:42]: I'm just move from there
https://github.com/apache/ignite/blob/6b6d82a3725a7202ff1effdf905e2463b084cbc7/modules/core/src/main/java/org/apache/ignite/internal/visor/verify/VisorValidateIndexesTaskArg.java#L128

File: modules/indexing/src/main/java/org/apache/ignite/internal/visor/verify/ValidateIndexesClosure.java [hunk 4]
```diff
@@ -206,53 +230,55 @@ private VisorValidateIndexesJobResult call0() {
...
         Map<Integer, IndexIntegrityCheckIssue> integrityCheckResults = integrityCheckIndexesPartitions(grpIds);
 
+        GridQueryProcessor qryProcessor = ignite.context().query();
+        IgniteH2Indexing h2Indexing = (IgniteH2Indexing)qryProcessor.getIndexing();
+
         for (Integer grpId : grpIds) {
             CacheGroupContext grpCtx = ignite.context().cache().cacheGroup(grpId);
 
-            if (grpCtx == null || integrityCheckResults.containsKey(grpId))
+            if (isNull(grpCtx) || integrityCheckResults.containsKey(grpId))
```
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/indexing/src/main/java/org/apache/ignite/internal/visor/verify/ValidateIndexesClosure.java:249]: I know all about it, but it's my style, besides java will eventually remove it, there's no overhead in it.

File: modules/indexing/src/main/java/org/apache/ignite/internal/visor/verify/ValidateIndexesClosure.java [hunk 5]
```diff
@@ -412,7 +476,7 @@ private IndexIntegrityCheckIssue integrityCheckIndexPartition(CacheGroupContext
         finally {
             integrityCheckedIndexes.incrementAndGet();
 
-            printProgressIfNeeded("Current progress of ValidateIndexesClosure: checked integrity of "
+            printProgressIfNeeded(() -> "Current progress of ValidateIndexesClosure: checked integrity of "
```
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/indexing/src/main/java/org/apache/ignite/internal/visor/verify/ValidateIndexesClosure.java:479]: Yes, plus there is a lambda cache and string is not created if not needed.

File: modules/indexing/src/main/java/org/apache/ignite/internal/visor/verify/ValidateIndexesClosure.java [hunk 6]
```diff
@@ -206,53 +230,55 @@ private VisorValidateIndexesJobResult call0() {
 
         List<Future<Map<PartitionKey, ValidateIndexesPartitionResult>>> procPartFutures = new ArrayList<>();
         List<Future<Map<String, ValidateIndexesPartitionResult>>> procIdxFutures = new ArrayList<>();
+        List<T3<CacheGroupContext, GridDhtLocalPartition, Future<CacheSize>>> cacheSizeFutures = new ArrayList<>();
+        List<T3<GridCacheContext, Index, Future<T2<Throwable, Long>>>> idxSizeFutures = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/indexing/src/main/java/org/apache/ignite/internal/visor/verify/ValidateIndexesClosure.java:234]: Can I not do this?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sk0x50) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java:2911]: Ok, got it!

**Pull request closed without merging**